Made http and native remoting addresses overridable via properties

mclaudio76 opened the pull request:
Added two properties errai.jboss.httpremotingaddress and errai.jboss.nativeremotingaddress to ovverride remoting addresses - may be useful if for any reason one needs to bind wildfly instance to different remoting port than 9999 and 9990.

mbarkley commented on errai-cdi/jboss/src/main/java/org/jboss/errai/cdi/server/as/JBossServletContainerAdaptor.java:
```diff
@@ -34,8 +34,8 @@
 
-  private static final String NATIVE_CONTROLLER_PATH = "remote://localhost:9999";
-  private static final String HTTP_CONTROLLER_PATH = "http-remoting://localhost:9990";
+  private static String NATIVE_CONTROLLER_PATH = "remote://localhost:9999";
```
By convention only constants use upper-case letters and underscores. Since these fields are now mutable the names should be changed to `nativeControllerPath` and `httpControllerPath`.

Also, since you assign values to these fields in the constructor below, these should probably become instance fields (i.e. not `static`).

mbarkley commented on errai-cdi/jboss/src/main/java/org/jboss/errai/cdi/server/as/JBossServletContainerAdaptor.java:
```diff
@@ -52,15 +52,51 @@
+    
     logger.branch(Type.INFO, "Starting container initialization...");
+    // Overrides remoting address if and only if an ovverride is passed.
+    if(httpRemotingAddress != null && !httpRemotingAddress.trim().equalsIgnoreCase(HTTP_CONTROLLER_PATH)) {
```
Minor formatting issue: we include a space between the `if` and the condition.

mbarkley commented on errai-cdi/jboss/src/main/java/org/jboss/errai/cdi/server/as/JBossServletContainerAdaptor.java:
```diff
@@ -52,15 +52,51 @@
+    	HTTP_CONTROLLER_PATH = httpRemotingAddress;
+    }
+
+    if(nativeRemotingAddress != null && !nativeRemotingAddress.trim().equalsIgnoreCase(NATIVE_CONTROLLER_PATH)) {
```
Minor formatting issue: we include a space between the if and the condition.

mbarkley commented on errai-cdi/jboss/src/main/java/org/jboss/errai/cdi/server/gwt/JBossLauncher.java:
```diff
@@ -38,6 +38,10 @@
   private final String JBOSS_JAVA_OPTS_PROPERTY = "errai.jboss.javaopts";
   private final String TMP_CONFIG_FILE = "standalone-errai-dev.xml";
+  
+  // Added 02/13/2016
```
Was this comment added by your IDE? It would be good to remove since it is possible to find when these fields were added through Git.

mbarkley commented on errai-cdi/jboss/src/main/java/org/jboss/errai/cdi/server/gwt/JBossLauncher.java:
```diff
@@ -45,13 +49,17 @@
     // Get properties
-    final String DEBUG_PORT = System.getProperty(JBOSS_DEBUG_PORT_PROPERTY, "8001");
-    final String TEMPLATE_CONFIG_FILE = System.getProperty(TEMPLATE_CONFIG_FILE_PROPERTY, "standalone-full.xml");
+    final String DEBUG_PORT 		     = System.getProperty(JBOSS_DEBUG_PORT_PROPERTY, "8001");
```
I see you've added whitespace here to align the equals signs, but this line is off by a couple spaces. This isn't something we typically do in Errai code so it might be best to remove this formatting.

mbarkley commented on errai-cdi/jboss/src/main/java/org/jboss/errai/cdi/server/gwt/JBossLauncher.java:
```diff
@@ -45,13 +49,17 @@
     final String CLASS_HIDING_JAVA_AGENT = System.getProperty(CLASS_HIDING_JAVA_AGENT_PROPERTY);
     String JAVA_OPTS = System.getProperty(JBOSS_JAVA_OPTS_PROPERTY, "");
+    
+    // Added new properties to allow overriding of default ones.
```
No need for this comment. Your code is quite clear :)

mbarkley commented on errai-cdi/jboss/src/main/java/org/jboss/errai/cdi/server/gwt/JBossLauncher.java:
```diff
@@ -45,13 +49,17 @@
     String JAVA_OPTS = System.getProperty(JBOSS_JAVA_OPTS_PROPERTY, "");
+    
+    // Added new properties to allow overriding of default ones.
+    final String HTTP_REMOTING_ADDRESS      =System.getProperty(JBOSS_HTTP_REMOTING_ADDRESS, "http-remoting://localhost:9990");
```
Please format this line and the one below so that there is a single space on either side of the equals symbol.

mbarkley commented:
Hi @mclaudio76 

Thank you for sending this PR. In spite of all my nit-picky in-line comments, this looks like a good change and I should be able to merge this after some superficial changes :-)

In my in-line comments I have pointed out stylistic and formatting issues. Please take a look at these and make the necessary changes. Additionally, it would be good to squash your changes into a single commit ([this](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#Squashing-Commits) article gives an explanation of how to do this if you are unfamiliar -- but this step I can do for you if you have issues).

Cheers.

mclaudio76 commented:
Thanks Max for your reply and hints... I hope I had fixed all issues with indentation (I also removed unnecessary comments). About "squashing corrections" in a single commit.. well, not sure I've done it right, I've not GIT client installed - I use EGIT in eclipse, can't reproduce steps explained in the article you linked, I proceeded more _ad sensum_. So, sorry for any trouble.
Can you gently tell me if it's now ok ?

mbarkley commented:
Everything looks good now. I squashed everything into a single commit for you.

[Here](https://github.com/errai/errai/commit/aea9285da2989fb22be688dc13b930b31c21daf9) is the commit on master, and [here](https://github.com/errai/errai/commit/b09a860dcac16800bd9c54b7d192673b8dd998fc) is the commit on 3.2.

Thanks again for your contribution!

mclaudio76 commented:
Thanks for helping me with commit..
